build.gradle

```diff
@@ -88,6 +102,36 @@ configurations
     integrationTestRuntime.extendsFrom testRuntime
 }
 
+protobuf {
+
+    generatedFilesBaseDir = 'gen/' 
+
+    protoc {
+        artifact = 'com.google.protobuf:protoc:2.6.1'
+    }
+
+    generateProtoTasks {
+        all().each { task ->
+            if (task.name == "generateProto") {
+                task.outputs.upToDateWhen {
+                    new File(generatedFilesBaseDir).exists()
+                }
+            }
+        }
+
+        task deleteGeneratedProto {
+            delete generatedFilesBaseDir
+        }
+
+        task recompileProto {}
+        recompileProto.dependsOn(deleteGeneratedProto)
+        recompileProto.dependsOn(generateProto)
+
+        compileJava.dependsOn(generateProto)
```

Comment:
Could you figure out what gradle tasks Eclipse's gradle plugin calls? Once we know that we could just add `someTask.dependeOn(recompileProto)`?

Reply:
I will look into it. Some cursory googling I did the other day turned up nothing. I may have to dive deep into documentation.